test/unit/eigensolver/mc/test_backtransformation.cpp

```diff
@@ -0,0 +1,179 @@
+}
+
+template <class T>
+void set_zero(Matrix<T, Device::CPU>& mat) {
+  dlaf::matrix::util::set(mat, [](auto&&) { return static_cast<T>(0.0); });
+}
+
+template <class T>
+void getTau(T& tau, T dotprod, BaseType<T> /*tau_i*/) {
+  tau = static_cast<T>(2.0) / dotprod;
+}
+
+template <class T>
+void getTau(std::complex<T>& tau, T dotprod, BaseType<T> tau_i) {
+  tau = {(static_cast<T>(1.0) + sqrt(static_cast<T>(1.0) - dotprod * tau_i * tau_i)) / dotprod, tau_i};
+}
+
+template <class T>
+void testBacktransformationEigenv(SizeType m, SizeType n, SizeType mb, SizeType nb) {
+  LocalElementSize sizeC(m, n);
+  TileElementSize blockSizeC(mb, nb);
+  Matrix<T, Device::CPU> mat_c(sizeC, blockSizeC);
+  dlaf::matrix::util::set_random(mat_c);
+
+  LocalElementSize sizeV(m, m);
+  TileElementSize blockSizeV(mb, mb);
+  Matrix<T, Device::CPU> mat_v(sizeV, blockSizeV);
+  dlaf::matrix::util::set_random(mat_v);
+
+  int tottaus;
+  if (m < mb || m == 0 || n == 0)
+    tottaus = 0;
+  else
+    tottaus = (m / mb - 1) * mb + m % mb;
+
+  if (tottaus > 0) {
+    // Impose orthogonality: Q = I - v tau v^H is orthogonal (Q Q^H = I)
+    // leads to tau = [1 + sqrt(1 - vH v taui^2)]/(vH v) for real
+    LocalElementSize sizeTau(m, 1);
+    TileElementSize blockSizeTau(1, 1);
+    Matrix<T, Device::CPU> mat_tau(sizeTau, blockSizeTau);
+
+    // Reset diagonal and upper values of V
+    MatrixLocal<T> v({m, m}, blockSizeV);
+    for (const auto& ij_tile : iterate_range2d(v.nrTiles())) {
+      // copy only the panel
+      const auto& source_tile = mat_v.read(ij_tile).get();
+      copy(source_tile, v.tile(ij_tile));
+      if (ij_tile.row() <= ij_tile.col()) {
+        tile::set0<T>(v.tile(ij_tile));
+      }
+      else if (ij_tile.row() == ij_tile.col() + 1) {
+        tile::laset<T>(lapack::MatrixType::Upper, 0.f, 1.f, v.tile(ij_tile));
+      }
+    }
+
+    // Create C local
+    MatrixLocal<T> c({m, n}, blockSizeC);
+    for (const auto& ij_tile : iterate_range2d(c.nrTiles())) {
+      // copy only the panel
+      const auto& source_tile = mat_c.read(ij_tile).get();
+      copy(source_tile, c.tile(ij_tile));
+    }
+
+    // TODO: creating a whole matrix, solves issues on the last tile when m%mb != 0 ==> find out how to
+    // use only a panel
+    LocalElementSize sizeT(tottaus, tottaus);
+    TileElementSize blockSizeT(mb, mb);
+    Matrix<T, Device::CPU> mat_t(sizeT, blockSizeT);
+    set_zero(mat_t);
+
+    common::internal::vector<hpx::shared_future<common::internal::vector<T>>> taus;
+
+    MatrixLocal<T> tausloc({tottaus, 1}, {mb, mb});
+    auto tau_rows = tausloc.nrTiles().rows();
+
+    auto nt = 0;
+    for (SizeType i = 0; i < tau_rows; ++i) {
+      common::internal::vector<T> t_tile;
+      auto seed = 10000 * i / mb + 1;
+      dlaf::matrix::util::internal::getter_random<BaseType<T>> random_value(seed);
+      for (SizeType t = 0; t < mb && nt < tottaus; ++t) {
+        const GlobalElementIndex v_offset{i * mb + t, i * mb + t};

```

Comment:
Offset is wrong:
With the new loop indices it should be: {j + mb, j}
(with the old notation {(i+1) * mb + t, i * mb + t})